seacrowd/sea_datasets/total_defense_meme/total_defense_meme.py

```diff
@@ -0,0 +1,277 @@
+        """Returns SplitGenerators."""
+        # download image from gdrive
+        output_dir = Path.cwd() / "data" / _DATASETNAME
+        output_dir.mkdir(parents=True, exist_ok=True)
+        output_file = output_dir / f"{_DATASETNAME}.zip"
+        if not output_file.exists():
+            gdown.download(_URLS["image"], str(output_file), fuzzy=True)
+        else:
+            print(f"File already downloaded: {str(output_file)}")
+        # extract image data
+        image_dir = Path(dl_manager.extract(output_file)) / "TD_Memes"
+
+        # download annotations
+        annotation_path = Path(dl_manager.download(_URLS["annotations"]))
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "image_dir": image_dir,
+                    "annotation_file": annotation_path,
+                },
+            ),
+        ]
+
+    def _generate_examples(self, image_dir: Path, annotation_file: Path) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        # load annotation
+        with open(annotation_file, "r", encoding="utf-8") as file:
+            annotation = json.load(file)
+
+        # get unique image names
+        image_names = sorted(
+            list(
+                set(annotation["Non_Memes"])
+                | set(annotation["Non_SG_Memes"])
+                | set(annotation["SG_Memes"])
+            )
+        )
+
+        # annotation data is a list of dict, instead of dict of image names
+        def get_value(image_name, list_of_dicts):
+            for dictionary in list_of_dicts:
+                if image_name in dictionary:
+                    return dictionary[image_name]
+            return None
+
+        key = 0
+        for image_name in image_names:
+            # assert image exist in directory
+            assert (image_dir / image_name).exists(), f"Image {image_name} not found"
+            image_path = str(image_dir / image_name)
+
+            # get categories, can be multiple
+            categories = []
+            if image_name in annotation["Non_Memes"]:
+                categories.append("Non_Memes")
+            if image_name in annotation["Non_SG_Memes"]:
+                categories.append("Non_SG_Memes")
+            if image_name in annotation["SG_Memes"]:
+                categories.append("SG_Memes")
+
+            # get attributes
+            text = get_value(image_name, annotation["Text"])
+            tags = get_value(image_name, annotation["Tags"])
+            raw_pillar_stances = get_value(image_name, annotation["Pillar_Stances"])
+
+            # process pillar stances
+            pillar_stances = []
+            if raw_pillar_stances:
+                for pillar, stances in raw_pillar_stances:
+                    category = pillar.split(" ")[0]
+                    pillar_stances.append({"category": category, "stance": stances})
```

Comment:
Could you please make this `"stance"` --> `"agreed_stances"` and add another variable `"all_stances"`?

The `all_stances` labels have all the stance annotations while the `agreed_stances` labels are the "correct labels" based on [this processing detail in Section 3.3 Quality Control Measures from the paper](https://arxiv.org/pdf/2305.17911.pdf).

```
Lastly, the annotators annotate the meme’s stances towards the assigned pillars: support, against, or neutral. To ensure the reliability of the dataset, each meme is annotated by two annotators. If the disagreements contain similar opinions, the overlap annotations will be considered correct labels. However, if there are disagreements with entirely different perspectives, a third annotator will be brought in to provide an additional annotation for the meme. The overlapping annotations between at least two annotators will then be considered the correct labels. In the extreme case where all three annotators have different opinions, the meme will be flagged and removed from the dataset.
```

Reply:
See my comment above.